File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    const visited = new Array(n).fill(false);
    
    const dfs = (i) => {
        if (!visited[i]) visited[i] = true;
        for (let j = 0; j < n; j++) {
            if (j !== i && computers[i][j] && !visited[j]) dfs(j);
        }
    }
    
    for (let i = 0; i < n; i++) {
        if (visited[i]) continue;
        dfs(i);
        answer++;
    }
    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        int[][] adj = computers;

        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                answer++;
                dfs(i, visited, adj);
            }
        }

        return answer;
    }

    private void dfs(int crnt, boolean[] visited, int[][] adj) {
        visited[crnt] = true;
        for (int i = 0; i < adj[crnt].length; i++) {
            if (adj[crnt][i] == 1 && !visited[i]) {
                dfs(i, visited, adj);
            }
        }
    }
}